src/components/comments/comment.resolver.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  Args,
+  Mutation,
+  Parent,
+  Query,
+  ResolveField,
+  Resolver,
+} from '@nestjs/graphql';
+import {
+  AnonSession,
+  ID,
+  IdArg,
+  ListArg,
+  LoggedInSession,
+  mapSecuredValue,
+  Session,
+} from '../../common';
+import { Loader, LoaderOf } from '../../core';
+import { SecuredUser, UserLoader } from '../user';
+import { CommentThreadLoader } from './comment-thread.loader';
+import { CommentLoader } from './comment.loader';
+import { CommentService } from './comment.service';
+import {
+  Comment,
+  CommentListInput,
+  CommentListOutput,
+  CommentThread,
+  CreateCommentInput,
+  CreateCommentOutput,
+  CreateCommentThreadInput,
+  CreateCommentThreadOutput,
+  DeleteCommentOutput,
+  UpdateCommentInput,
+  UpdateCommentOutput,
+} from './dto';
+
+@Resolver(Comment)
+export class CommentResolver {
+  constructor(private readonly service: CommentService) {}
+
+  @Mutation(() => CreateCommentOutput, {
+    description: 'create a comment',
+  })
+  async createComment(
+    @LoggedInSession() session: Session,
+    @Args('input') { comment: input }: CreateCommentInput
+  ): Promise<CreateCommentOutput> {
+    const comment = await this.service.create(input, session);
+    return { comment };
+  }
+
+  @Query(() => Comment, {
+    description: 'Look up a comment by ID',
+  })
+  async comment(
+    @IdArg() id: ID,
+    @Loader(CommentLoader) comments: LoaderOf<CommentLoader>
+  ): Promise<Comment> {
+    return await comments.load(id);
+  }
+
+  @Mutation(() => UpdateCommentOutput, {
+    description: 'Update an existing comment',
+  })
+  async updateComment(
+    @LoggedInSession() session: Session,
+    @Args('input') { comment: input }: UpdateCommentInput
+  ): Promise<UpdateCommentOutput> {
+    const comment = await this.service.update(input, session);
+    return { comment };
+  }
+
+  @Mutation(() => DeleteCommentOutput, {
+    description: 'Delete a comment',
+  })
+  async deleteComment(
+    @LoggedInSession() session: Session,
+    @IdArg() id: ID
+  ): Promise<DeleteCommentOutput> {
+    await this.service.delete(id, session);
+    return { success: true };
+  }
+
+  @Mutation(() => CreateCommentThreadOutput, {
+    description: 'Create a comment thread',
+  })
+  async createCommentThread(
```

Comment:
Instead of making the consumer call this only to turn around and call `createComment`. Let's merge the two. `createComment` can take a `resourceId` & optionally a `threadId`. If no threadId is given we can create the thread for them before creating the comment.
The output can be changed to have both the comment & the comment thread.